webhook/webhook.go

```diff
@@ -148,16 +142,26 @@ func New(
 func (ac *Webhook) Run(stop <-chan struct{}) error {
 	logger := ac.Logger
 	ctx := logging.WithLogger(context.TODO(), logger)
-	tlsConfig, caCert, err := configureCerts(ctx, ac.Client, &ac.Options)
+
+	// TODO(mattmoor): Separate out the certificate creation process and use listers
+	// to fetch this from the secret below.
+	serverKey, serverCert, caCert, err := getOrGenerateKeyCertsFromSecret(ctx, ac.Client, &ac.Options)
 	if err != nil {
-		logger.Errorw("could not configure admission webhook certs", zap.Error(err))
 		return err
 	}
 
 	server := &http.Server{
-		Handler:   ac,
-		Addr:      fmt.Sprintf(":%v", ac.Options.Port),
-		TLSConfig: tlsConfig,
+		Handler: ac,
+		Addr:    fmt.Sprintf(":%v", ac.Options.Port),
+		TLSConfig: &tls.Config{
+			GetCertificate: func(*tls.ClientHelloInfo) (*tls.Certificate, error) {
+				cert, err := tls.X509KeyPair(serverCert, serverKey)
```

Comment:
Do we want to cache this ? E.g. with once?

Reply:
No because eventually this is going to be synthesized from the secret pulled from the informer cache (probably the next change).